### DISCUSSION TITLE
Library documentation

### DISCUSSION THREAD

**Discussion by husqD** [KUL-RSDA/AquaCrop#325] [General]
Hello, 

I am trying to use the library but it is very difficult without documentation. 
Do you plan to create a guide to use the library?
It is not very usable without proper documentation.

Thanks

**Comment by mbechtold** [comment]
We assume this refers to using AquaCrop as a library to compile and use it within other source code, such as e.g. NASA’s Land Information System. Indeed, documentation of the library is still pending. We are currently still in the process of making the interfaces of the library more suitable for the coupling to data assimilation frameworks. Some modules are not well accessible yet since AquaCrop was originally developed as a standalone / GUI version. Once our structural changes are finished, the first documentation will be made. We cannot promise a date for this first documentation right now.
For the moment, tools like doxygen (https://www.doxygen.nl/) might be useful which produce documentation based on the source code including its comments.
